src/radical/pilot/agent/scheduler/flux.py

```diff
@@ -144,5 +144,49 @@ def work(self, tasks):
   #     return new_env
 
 
+    # --------------------------------------------------------------------------
+    #
+    def task_to_spec(self, task):
```

Comment:
I would propose to move it into LM (or at least to executor), but if to keep it here is just a temporary solution then ok

Reply:
Yes, I agree with that certainly.  If you don't mind though, let's clean this up after `env_isolation` merge.  I'll open a ticket to keep track of it.  Thanks!